controllers/utils/storageclass.go

```diff
@@ -0,0 +1,148 @@
+	cephFSName := fmt.Sprintf(defaultCephFSStorageClassTemplate, storageClusterName)
+
+	rbdSC, err := GetStorageClass(ctx, c, rbdName)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return nil, fmt.Errorf("error getting RBD StorageClass: %v", err)
+		}
+	} else {
+		defaultStorageClasses = append(defaultStorageClasses, rbdSC)
+	}
+
+	cephFSSC, err := GetStorageClass(ctx, c, cephFSName)
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return nil, fmt.Errorf("error getting CephFS StorageClass: %v", err)
+		}
+	} else {
+		defaultStorageClasses = append(defaultStorageClasses, cephFSSC)
+	}
+
+	return defaultStorageClasses, nil
+}
+
+func UpdateStorageClass(ctx context.Context, c client.Client, storageClass *storagev1.StorageClass) error {
+	if err := c.Update(ctx, storageClass); err != nil {
+		return fmt.Errorf("failed to update StorageClass %s: %v", storageClass.Name, err)
+	}
+	return nil
+}
+
+func GetStorageIdsForDefaultStorageClasses(ctx context.Context, c client.Client, scNamespacedName types.NamespacedName, spokeClusterName string) (map[CephType]string, error) {
+	storageClasses, err := GetDefaultStorageClasses(ctx, c, scNamespacedName.Name)
+
+	if err != nil {
+		return nil, fmt.Errorf("error occured while fetching the default storageclasses. %v", err)
+	}
+
+	storageIdMap := make(map[CephType]string)
+	for _, sc := range storageClasses {
+		storageClassType, err := GetStorageClassType(sc, scNamespacedName.Namespace)
+		if err != nil {
+			return nil, err
+		}
+		storageId, err := CalculateStorageId(ctx, c, sc, scNamespacedName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to calculate the storageId of storageclass %s. error: %v", sc.Name, err)
+		}
+		storageIdMap[storageClassType] = storageId
+	}
+
+	return storageIdMap, err
+}
+
+func GetStorageClassType(sc *storagev1.StorageClass, storageClusterNamespace string) (CephType, error) {
+	switch sc.Provisioner {
+	case fmt.Sprintf(RBDProvisionerTemplate, storageClusterNamespace):
+		return RBDType, nil
+
+	case fmt.Sprintf(CephFSProvisionerTemplate, storageClusterNamespace):
+		return CephFSType, nil
+
+	default:
+		return "", fmt.Errorf("unknown provisioner type: %s", sc.Provisioner)
+	}
+}
+
+func CalculateStorageId(ctx context.Context, c client.Client, sc *storagev1.StorageClass, storageClusterNamespacedName types.NamespacedName) (string, error) {
+	cephCluster, err := FetchCephCluster(ctx, c, storageClusterNamespacedName)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch ceph cluster: %v", err)
+	}
+
+	cephClusterFSID := cephCluster.Status.CephStatus.FSID
+	var storageId string
+
+	switch sc.Provisioner {
+	case fmt.Sprintf(RBDProvisionerTemplate, storageClusterNamespacedName.Namespace):
+		poolID, err := fetchPoolIdForCephBlockPool(ctx, c, sc, storageClusterNamespacedName.Namespace)
+		if err != nil {
+			return "", fmt.Errorf("failed to get pool ID: %v", err)
+		}
+		storageId = CreateUniqueName(cephClusterFSID, poolID, DefaultRadosNamespace)
```

Comment:
Could we align to use the same way to generate the storageID: https://github.com/red-hat-storage/ocs-operator/pull/3008#discussion_r1943627812

Reply:
Agree. Also we need to add tests to ensure we're generating exact same IDs for same inputs.

Reply:
Tests written